prov/sockets/src/sock_ep_msg.c

```diff
@@ -244,13 +244,12 @@ static int sock_ep_cm_getname(fid_t fid, void *addr, size_t *addrlen)
 	struct sock_ep *sock_ep = NULL;
 	struct sock_pep *sock_pep = NULL;
 
-	if (*addrlen == 0) {
+	if (*addrlen < sizeof(struct sockaddr_in)) {
```

Comment:
The original code is correct.  The address should be truncated if addrlen is too small.